### PR TITLE
LABS-1054 Mark the ingest table page dirty if the page is not reconciled to an empty page

### PR DESCRIPTION
**Pull request by quchenhao**
For the ingest table, if the page cannot be reconciled to an empty page, we have more work to do for garbage collection. Mark it as dirty to reconcile it again in the future.

### CODE REVIEW

File: src/btree/bt_split.c [hunk 1]
```diff
@@ -1414,6 +1414,9 @@ __split_multi_inmem(WT_SESSION_IMPL *session, WT_PAGE *orig, WT_MULTI *multi, WT
     if (!__wt_readgen_evict_soon(&orig_read_gen))
         __wt_atomic_store64(&page->read_gen, orig_read_gen);
 
+    if (F_ISSET(S2BT(session), WT_BTREE_GARBAGE_COLLECT))
```
Comment (korteland) [inline src/btree/bt_split.c:1417]: The title of the PR says it should only do this if the page is not reconciled to an empty page. Where is that check done? It doesn't seem to be in `__split_multi_inmem`, is there another part of the reconciliation code where this happens?

**Review by korteland — COMMENTED** [review]
One question - sorry it took me so long to get around to reviewing this.

File: src/btree/bt_split.c (hunk 1 above)
Comment (quchenhao) [reply src/btree/bt_split.c:1417]: Here we are reading the reconciled image to memory. We are only here when the page is not empty.

Comment (korteland) [reply src/btree/bt_split.c:1417]: Got it - thanks.

**Review by korteland — APPROVED** [review]

File: src/btree/bt_split.c (hunk 1 above)
Comment (pmacko86) [inline src/btree/bt_split.c:1417]: Could you please add a comment explaining why we need this?

**Review by pmacko86 — APPROVED** [review]
LGTM after you add a comment—thanks :)

**Comment by agorrod** [comment]
I guess this is OK, but I don't think it's critical for correctness that content is garbage collected. As such, I'm not sure we want to jump to this extreme to ensure content is visited again.

OTOH: Fine - as long as it's obvious what the code is doing, and we remember to be deliberate about it in the future, this will at least make testing easier, as content is aggressively garbage collected from the ingest table.

**Review by agorrod — COMMENTED** [review]
I left a general comment - I'm happy for this to be merged, but don't find the change compelling.

**Pull request merged**